src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -262,11 +262,11 @@
  * <p>
  * Alternatively, a memory access operation is <em>partially aligned</em> if it occurs at a memory address {@code A}
  * which is only compatible with the alignment constraint {@code B}; in such cases, access for anything other than the
- * {@code get} and {@code set} access modes will result in an {@code IllegalStateException}. If access is partially aligned,
+ * {@code get} and {@code set} access modes will result in an {@code IllegalArgumentException}. If access is partially aligned,
```

Comment:
I believe this and the following have always been "wrong" - e.g. ISE instead of IAE (which is what the code throws)